src/expr/expr.rs

```diff
@@ -18,19 +23,34 @@ pub enum BinOp {
 
 impl Expr {
     pub fn evaluate(&self) -> isize {
-        unimplemented!()
+        match self {
+            //references makes us not-sad here - note that &self is a reference.
+            &Expr::BinOp(ref left_box, BinOp::Plus, ref right_box) => (*left_box).evaluate() + (*right_box).evaluate(),
+            &Expr::BinOp(ref left_box, BinOp::Minus, ref right_box) => (*left_box).evaluate() - (*right_box).evaluate(),
+            &Expr::BinOp(ref left_box, BinOp::Times, ref right_box) => (*left_box).evaluate() * (*right_box).evaluate(),
+            &Expr::BinOp(ref left_box, BinOp::Over, ref right_box) => (*left_box).evaluate() / (*&right_box).evaluate(),
```

Comment:
It turns out the explicit dereferences aren't needed here.

Also, for what it's worth the type of `*left_box` here is `Box<Expr>`.